### PR TITLE
Structural changes

### PR DESCRIPTION
**Pull request by rkuchan**
This attempts to make the docs more accessible and discoverable.

The first commit adds details to the docs landing page (see below). It definitely needs work on formatting and how it's actually in there (please help!) but the content is there.
image

This also makes significant structural changes to the sidebar.
It makes the Tools section more about the actual tools for interacting with the database; makes the Write Protocols section all about the Line Protocol; and makes the Query Language section all about InfluxQL:

* Moves Tools to below Troubleshooting
* Moves the API reference from the Concepts section to Tools
* Moves Client Libraries from its own section to Tools
* Moves service plugins from Write Protocols to Tools
* Removes Telegraf, Chronograf, and Kapacitor from Tools
* Increases accessibility of the Sample Data by adding it to Query Language
* Moves Authentication and Authorization from Administration to Query Language

It fills out the index pages for the sections listed above to make the docs more accessible as well as help users distinguish between reference documentation and tutorial-style documentation.

Finally, it makes major content changes to the two write syntax pages. It renames them to Line Protocol Tutorial and Line Protocol Reference.

### CODE REVIEW

**Comment by beckettsean** [comment]
@kostasb can you deploy this branch to docs staging when you get a chance? I'm too lazy/incompetent to build the docs locally and I can't inspect this PR without the browser experience.

File: content/influxdb/v1.0/write_protocols/line_protocol_reference.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+| [Field set](/influxdb/v1.0/concepts/glossary/#field-set) | Required. Points must have at least one field. | All field key-value pairs for the point. | [Field keys](/influxdb/v1.0/concepts/glossary/#field-key) are strings. [Field values](/influxdb/v1.0/concepts/glossary/#field-value) can be floats, integers, strings, or booleans.
+| [Timestamp](/influxdb/v1.0/concepts/glossary/#timestamp) | Optional. InfluxDB uses the server's local nanosecond timestamp in UTC if the timestamp is not included with the point. | The timestamp for the data point. InfluxDB accepts one timestamp per point. | Unix nanosecond timestamp. Specify alternative precisions with the [HTTP API](/influxdb/v1.0/tools/api/#write).
+
+> **Performance tips:**
+>
+* Sort tags by key before sending them to the database.
+The sort should match the results from the
+[Go bytes.Compare function](http://golang.org/pkg/bytes/#Compare).
+>
+* Use the least precise
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_reference.md]: Use the coarsest precision?

File: content/influxdb/v1.0/write_protocols/line_protocol_reference.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+Line Protocol does not require users to escape the backslash character `\`.
+Users do not need to escape all other special characters.
+
+#### Example
+
+Write a point where the measurement is `"measurement with quo⚡️es and emoji"`, the tag key is `tag key with sp🚀ces`, the
+tag value is `tag,value,with"commas"`, the field key is `field_key\\\\` and the field value is `string field value, only " need be qu🍭ted`.
+
+```
+> INSERT "measurement\ with\ quo⚡️es\ and\ emoji",tag\ key\ with\ sp🚀ces=tag\,value\,with"commas" field_k\ey="string field value, only \" need be qu🍭ted"
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_reference.md]: The field key is screwed up. I think it should be `field_key\\\\=`

Also, shouldn't it be `"only " need be escaped"`, not `quoted`?

File: content/influxdb/v1.0/write_protocols/line_protocol_reference.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+For string field values use a backslash character `\` to escape:
+
+* double quotes `"`
+
+Line Protocol does not require users to escape the backslash character `\`.
+Users do not need to escape all other special characters.
+
+#### Example
+
+Write a point where the measurement is `"measurement with quo⚡️es and emoji"`, the tag key is `tag key with sp🚀ces`, the
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_reference.md]: I love this, btw. Totally ridiculous and awesome display of Unicode and escape. characters

File: content/influxdb/v1.0/write_protocols/line_protocol_reference.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+tag value is `tag,value,with"commas"`, the field key is `field_key\\\\` and the field value is `string field value, only " need be qu🍭ted`.
+
+```
+> INSERT "measurement\ with\ quo⚡️es\ and\ emoji",tag\ key\ with\ sp🚀ces=tag\,value\,with"commas" field_k\ey="string field value, only \" need be qu🍭ted"
+```
+
+### Additional Naming Guidelines
+
+`#` at the beginning of the line is a valid comment character for Line Protocol.
+InfluxDB will ignore all subsequent characters.
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_reference.md]: InfluxDB will ignore all subsequent characters until the next newline `\n`.

File: content/influxdb/v1.0/write_protocols/line_protocol_reference.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+See the [Tools](http://localhost:1313/influxdb/v1.0/tools/) section for how to
+write Line Protocol to the database.
+
+### Duplicate points
+
+A point is uniquely identified by the measurement name, tag set, and timestamp.
+If you submit Line Protocol with the same measurement, tag set, and timestamp,
+but with a different field set, the field set becomes the union of the old
+field set and the new field set, where any ties go to the new field set.
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_reference.md]: ...where any conflicts favor the new field set. (always weird to edit my own sentences months later)

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+newversionredirect: /write_protocols/line/
+menu:
+  influxdb_1:
+    weight: 0
+    parent: write_protocols
+---
+
+InfluxDB's Line Protocol is a text based format for writing points to the
+database.
+Points must be in Line Protocol format for InfluxDB to successfully parse and
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: Maybe "Points written to the `/write` endpoint"? After all, Graphite, OpenTSDB, and CollectD plugins are part of the database but don't receive line protocol.

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+write points.
+
+This page introduces the syntax of Line Protocol using fictional temperature
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: Maybe: "Using fictional temperature data, this page introduces the syntax of Line Protocol. It covers:"

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+
+In the example, the field set consists of one field: `temperature=82`.
+Adding another field (`bug_concentration=98`) to the example looks like this:
+```
+weather,location=us-midwest temperature=82,bug_concentration=98 1465839830100400200
+```
+
+### Whitespace II
+
+Separate the field set and, if you're including a timestamp with your data
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: Separate the field set and the optional timestamp with a whitespace.

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+point, the timestamp with a whitespace.
+The whitespace is required in Line Protocol if you're including a timestamp
+(see the next section for more on timestamps).
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: Strike the parenthetical, I think. It's literally just below this line.

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+In the example, the timestamp is `1465839830100400200` (that's
+`2016-06-13T17:43:50.1004002Z` in RFC3339 format).
+The Line Protocol below is the same data point but without the timestamp.
+When InfluxDB writes it to the database it uses your server's
+local timestamp instead of `2016-06-13T17:43:50.1004002Z`.
+```
+weather,location=us-midwest temperature=82
+```
+
+It is possible to specify the precision (that is, if the timstamp is in hours,
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: It is possible to specify timestamps with a precision other than nanoseconds, such as microseconds, millseconds, or seconds. <#SB - we may not support minutes or hours much longer and they seem pretty useless>

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 11]
```diff
@@ -0,0 +1,418 @@
...
+minutes, seconds, and so on) of the timestamp with the HTTP API.
+We recommend using the least precise precision possible as this can result in
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: Maybe "We recommend using the coarsest precision possible..."

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+
+    Store the field value `82` as an integer:
+    ```
+weather,location=us-midwest temperature=82i 1465839830100400200
+    ```
+
+* Strings - double quote string field values (more on quoting in Line Protocol
+[below](#quoting)).
+
+    Store the field value `too hot` as a string:
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: I'd rather use `warm day` than `too hot`. The latter can come off as SF-centric. 82 degrees isn't a particularly hot day for a good chunk of the country and the world.

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 13]
```diff
@@ -0,0 +1,418 @@
...
+weather,location=us-midwest temperature="too\"hot\"" 1465839830100400200
+    ```
+
+Line Protocol does not require users to escape the backslash character `\`.
+All other special characters also do not require escaping.
+For example, Line Protocol handles emojis with no problem:
+
+```
+> INSERT we⛅️ther,location=us-midwest temper🔥ture=82 1465839830100400200
+>
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: I think show the `SELECT` statement and the output, too. One, to prove the point was valid and persisted, and two, to show that the CLI and the query engine have no issues with Unicode, either.

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 14]
```diff
@@ -0,0 +1,418 @@
...
+
+You can also use the CLI to
+[import](/influxdb/v1.0/tools/shell/#import-data-from-a-file-with-import) Line
+Protocol from a file.
+
+There are several ways to write data to InfluxDB.
+See the [Tools](http://localhost:1313/influxdb/v1.0/tools/) section for more
+on the [HTTP API](/influxdb/v1.0/tools/api/#write), the
+[CLI](/influxdb/v1.0/tools/shell/), and other possibilities.
+
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: Maybe also a link to https://docs.influxdata.com/influxdb/v1.0/write_protocols/ to at least hint at Graphite, CollectD, and OpenTSDB endpoints.

Should have a link to the UDP API (https://docs.influxdata.com/influxdb/v1.0/write_protocols/udp/) alongside the HTTP one.

File: content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md [hunk 15]
```diff
@@ -0,0 +1,418 @@
...
+### Duplicate points
+
+A point is uniquely identified by the measurement name, tag set, and timestamp.
+If you submit Line Protocol with the same measurement, tag set, and timestamp,
+but with a different field set, the field set becomes the union of the old
+field set and the new field set, where any ties go to the new field set.
```
Comment (beckettsean) [inline content/influxdb/v1.0/write_protocols/line_protocol_tutorial.md]: ...where any conflicts favor the new field set.

**Comment by beckettsean** [comment]
@rkuchan great work on the Reference and Tutorial! Some notes but otherwise ready to ship!

**Comment by kostasb** [comment]
@rkuchan @beckettsean 

staging rk-structural-changes: http://docs.staging.influxdata.com:31001/

**Comment by rkuchan** [comment]
Thanks @beckettsean!

**Comment by beckettsean** [comment]
> Moves Tools to below Troubleshooting

I like it, but I think the CLI should be at the top of the Tools section.

> Moves the API reference from the Concepts section to Tools

Makes a lot more sense there, I think.

> Moves Client Libraries from its own section to Tools

Good move, we need to eliminate that orphaned section, anyway.

> Moves service plugins from Write Protocols to Tools

Also a good move. Users who need them can find them, but they aren't distracting from the main API.

> Removes Telegraf, Chronograf, and Kapacitor from Tools

Very good move. They were wasted links with the other nav tools already available. 

> Increases accessibility of the Sample Data by adding it to Query Language

I like this move, too, but it shouldn't be at the top of the list. I think the bottom of the list is the right place for visibility.

> Moves Authentication and Authorization from Administration to Query Language

I also like this move a lot. And the new Ports section under Admin is a great way to keep the info but get it out of the API docs.

I'm a huge fan of these changes, just a few minor suggestions.

**Comment by rkuchan** [comment]
Thank you, @beckettsean! I just made all the requested changes.

**Comment by beckettsean** [comment]
@alexpaxton @benstronaut we can't wait any longer to push these changes, we really need to get them live. We're merging this now, but the landing page does still need design work.

**Pull request merged**